spec/integration/api/v1/config_spec.rb

```diff
@@ -26,6 +26,10 @@ def app()
         {'name' => 'pool1', 'size' => 5, 'template' => 'templates/pool1', 'clone_target' => 'default_cluster'},
         {'name' => 'pool2', 'size' => 10}
       ],
+      pools_at_startup: [
```

Comment:
I was trying to implement this and am now realizing I am not sure what you are suggesting.

Reply:
I was thinking something adding to the existing context 'when no config is given' because it calls Vmpooler.config which is the code block where the Marshall load is happening. Then its easy to swap for .clone and see if it works to replace Marshall load (it does not as you also found out, because its an array and you need a deep clone for the array)
Meaning it actually tests the code block here https://github.com/puppetlabs/vmpooler/blob/master/lib/vmpooler.rb#L30

So the spec would add something like this:
```
    context 'when no config is given' do
      it 'keeps a copy of the original pools at startup' do
        Dir.chdir(fixtures_dir) do
          configuration = Vmpooler.config
          expect(configuration[:pools]).to eq(configuration[:pools_at_startup])
        end
      end
      it 'the copy is a separate object and not a reference' do
        Dir.chdir(fixtures_dir) do
          configuration = Vmpooler.config
          configuration[:pools][0]['template'] = 'sam'
          expect(configuration[:pools]).not_to eq(configuration[:pools_at_startup])
        end
      end
    end
```

Reply:
Thanks! Added.